src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -189,11 +189,38 @@ private function flatten(array $array, array &$result, $keySeparator, $parentKey
             if (\is_array($value)) {
                 $this->flatten($value, $result, $keySeparator, $parentKey.$key.$keySeparator);
             } else {
-                $result[$parentKey.$key] = $value;
+                $result[$parentKey.$key] = $this->cleanValueByType($value);
             }
         }
     }
 
+    /**
+     * Ensure an actual value is used instead of a blank value when dealing
+     * with true, false and null (like a nullable bool in a database)
+     *
+     * @param string $value
+     *
+     * @return string
+     */
+    private function cleanValueByType($value)
+    {
+        $clean = $value;
+
+        if (false === $value) {
```

Comment:
Maybe could you use a `switch` instead of these `else/if`?

Reply:
I tried a switch, but it returned unexpected results because it only uses loose comparisons (like `null` evaluating as `false`, or `0` evaluating as `null`). With the triple equals it's okay though.

Reply:
Indeed! You're right.